cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiff.java

```diff
@@ -0,0 +1,214 @@
+            + "|.*-909[1-4]\\.sasl\\.enabled\\.mechanisms"
+            + "|advertised\\.listeners"
+            + "|zookeeper\\.connect"
+            + "|super\\.users"
+            + "|broker\\.rack)$");
+
+    public KafkaBrokerConfigurationDiff(Map<ConfigResource, Config> current, ConfigMap desired, KafkaVersion kafkaVersion, int brokerId) {
+        this.current = current;
+        this.diff = emptyKafkaConf(); // init
+        this.desired = desired;
+        this.kafkaVersion = kafkaVersion;
+        this.brokerId = brokerId;
+        Config brokerConfigs = this.current.get(new ConfigResource(ConfigResource.Type.BROKER, Integer.toString(brokerId)));
+        if (brokerConfigs != null) {
+            this.currentEntries = brokerConfigs.entries();
+            this.diff = computeDiff();
+        }
+    }
+
+    private KafkaConfiguration emptyKafkaConf() {
+        Map<String, Object> conf = new HashMap<>();
+        return new KafkaConfiguration(conf.entrySet());
+    }
+
+
+    private void fillPlaceholderValue(Map<String, String> orderedProperties, String placeholder, String value) {
+        orderedProperties.entrySet().forEach(entry -> {
+            entry.setValue(entry.getValue().replaceAll("\\$\\{" + Pattern.quote(placeholder) + "\\}", value));
+        });
+    }
+
+    public KafkaConfiguration getDiff() {
+        return this.diff;
+    }
+
+    public boolean isDesiredPropertyDefaultValue(String key) {
+        Optional<ConfigEntry> entry = currentEntries.stream().filter(configEntry -> configEntry.name().equals(key)).findFirst();
+        if (entry.isPresent()) {
+            return entry.get().isDefault();
+        }
+        return false;
+    }
+
+    public boolean cannotBeUpdatedDynamically() {
+        if (diff == null) {
+            return false;
+        } else return diff.anyReadOnly(kafkaVersion)
+                || !diff.unknownConfigs(kafkaVersion).isEmpty()
+                || diff.containsListenersChange();
+    }
+
+    /**
+     * @return Map object which is used for dynamic configuration of kafka broker
+     */
+    public Map<ConfigResource, Collection<AlterConfigOp>> getUpdatedConfig() {
+        return updated;
+    }
+
+    private boolean isIgnorableProperty(String key) {
+        return IGNORABLE_PROPERTIES.matcher(key).matches();
+    }
+
+    /**
+     * Computes diff between two maps. Entries in IGNORABLE_PROPERTIES are skipped
+     * @return KafkaConfiguration containing all entries which were changed from current in desired configuration
+     */
+    public KafkaConfiguration computeDiff() {
+        Map<String, String> currentMap = new HashMap<>();
+        Map<String, String> difference = new HashMap<>();
+
+        Collection<AlterConfigOp> updatedCE = new ArrayList<>();
+
+        currentEntries.stream().forEach(e -> {
+            currentMap.put(e.name(), e.value());
+        });
+
+        Map<String, String> desiredMap = ModelUtils.configMap2Map(desired, "server.config");
```

Comment:
Let's not get into the habit of using `2` to mean `to`. That way lies the madness of coding in textspeak.